packages/client/src/components/interface/Navigation.tsx

```diff
@@ -706,25 +707,25 @@ export const NavigationView = (props: IFullProps) => {
                 icon={() => (
                   <Icon name={'Star'} color={'yellow'} fill={'yellow'}></Icon>
                 )}
-                id={`bookmarked_advanced_search_${bookmarkResult.searchId}`}
-                label={bookmarkResult.name}
+                id={`bookmarked_advanced_search_${bookmarkResult?.searchId}`}
+                label={bookmarkResult?.name}
                 disabled={
-                  advancedSearchParams.searchId === bookmarkResult.searchId &&
+                  advancedSearchParams.searchId === bookmarkResult?.searchId &&
                   props.location.pathname === ADVANCED_SEARCH_RESULT
                 }
                 onClick={() => {
                   const filteredParam = omit(
-                    bookmarkResult.parameters,
+                    bookmarkResult?.parameters,
                     '__typename'
                   ) as IAdvancedSearchParamState
                   setAdvancedSearchParam({
                     ...filteredParam,
-                    searchId: bookmarkResult.searchId
+                    searchId: bookmarkResult?.searchId
                   })
                   goToAdvancedSearchResultAction()
                 }}
                 isSelected={
-                  advancedSearchParams.searchId === bookmarkResult.searchId &&
+                  advancedSearchParams.searchId === bookmarkResult?.searchId &&
```

Comment:
Good point! I tried to fix the type errors after I changed the type to the User. I have to change this back to where it was. 

I'm sorry for the size of this PR, my goal was to remove these custom frontend types: `IUserDetails, IAvatar, IGQLLocation, IIdentifier ` and replace them with generated gateway types. After the changes I had to fix a lot of type errors which fails from the unit tests.

Reply:
Okay yeah I kinda get it. Nothing we can do about the PR size I guess :sweat_smile:

Reply:
Yea you are right, the missing fields from the getUser query are `id` and `device`. Should I bring the `IUserDetails` type back and change it back to the way it was or can we add the missing fields to the query so we could keep using the User type. What you think?

It would be nice to use the `User` type because it has fewer optional fields.